internal/pkg/consul/client.go

```diff
@@ -45,13 +46,24 @@ type consulClient struct {
 	healthCheckInterval string
 }
 
+func getLastRune(s string) string {
+	j := len(s)
+	_, size := utf8.DecodeLastRuneInString(s[:j])
+	j -= size
+	return s[j:]
+}
+
 // Create new Consul Client. Service details are optional, not needed just for configuration, but required if registering
 func NewConsulClient(registryConfig types.Config) (*consulClient, error) {
 
 	client := consulClient{
 		serviceKey:     registryConfig.ServiceKey,
 		consulUrl:      registryConfig.GetRegistryUrl(),
-		configBasePath: registryConfig.Stem + registryConfig.ServiceKey + "/",
+		configBasePath: registryConfig.Stem + registryConfig.ServiceKey,
+	}
+
+	if getLastRune(client.configBasePath) != "/" {
```

Comment:
Simpler to use `if client.configBasePath[len(client.configBasePath)-1:] != "/"`  to check last character rather than right a function.

Reply:
Are we guaranteed the final character will always be exactly one byte?

Reply:
After some more thought, the comparison would be fine because UTF-8 doesn't use ASCII values in non-ASCII characters.  It does need a check against an empty string, though.  Updated PR.